### PR TITLE
Create Clean Arh, Create DAL

### PR DESCRIPTION
**Pull request by rokiiiiii**

### CODE REVIEW

**Comment by rokiiiiii** [comment]
У меня возникает ошибка, когда я обращаюсь к DbContext
image
И возникает ошибка при создании миграции 
image

**Comment by rokiiiiii** [comment]
Ошибку с миграция, пофиксил

File: SellCar/SellCar.Domain/Models/Ads.cs [hunk 1]
```diff
@@ -10,6 +10,10 @@ public class Ads
         public int UserId { get; set; }
         public int CarId { get; set; }
         public DateTime DateCreate { get; set; }
-      
+
+        public string Description { get; set; }
+
+        public int price { get; set; }
```
Comment (VadzimPapko) [inline SellCar/SellCar.Domain/Models/Ads.cs:16]: wrong Property naming

File: SellCar/SellCar.Service/Implementations/CarService.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        {
+            _carRepository = carRepository;
+        }
+
+        public async Task<IBaseResponse<Car>> GetCar(int id)
+        {
+            var baseResponse = new BaseResponse<Car>();
+            try
+            {
+                var user = await _carRepository.Get(id);
```
Comment (VadzimPapko) [inline SellCar/SellCar.Service/Implementations/CarService.cs:30]: why this one is called user?

File: SellCar/SellCar.Service/Implementations/CarService.cs [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+            }
+        }
+
+        public async Task<IBaseResponse<bool>> DeleteCar(int id)
+        {
+            var baseResponse = new BaseResponse<bool>();
+            try
+            {
+                var user = await _carRepository.Get(id);
```
Comment (VadzimPapko) [inline SellCar/SellCar.Service/Implementations/CarService.cs:81]: same

File: SellCar/SellCar.Service/Implementations/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            }
+        }
+
+
+    }
+}
+
+
+
+
```
Comment (VadzimPapko) [inline SellCar/SellCar.Service/Implementations/UserService.cs:188]: delete unused empty lines for all C# files

File: SellCar/SellCar.Service/Intrefaces/ICarSevice.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SellCar.Service.Intrefaces
+{
+    public interface ICarService
+    {
+
```
Comment (VadzimPapko) [inline SellCar/SellCar.Service/Intrefaces/ICarSevice.cs:16]: delete line

File: SellCar/SellCar.Service/Intrefaces/ICarSevice.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        public Task<IBaseResponse<CarViewModel>> CreateCar(CarViewModel carViewModel);
+        public Task<IBaseResponse<Car>> Edit(int id, CarViewModel model);
+        public Task<IBaseResponse<bool>> DeleteCar(int id);
+        public Task<IBaseResponse<Car>> GetByModel(string name);
+        public Task<IBaseResponse<IEnumerable<Car>>> GetCars();
+        public Task<IBaseResponse<Car>> GetCar(int id);
+
+
```
Comment (VadzimPapko) [inline SellCar/SellCar.Service/Intrefaces/ICarSevice.cs:24]: delete lines

File: SellCar/SellCar.DAL/Repositories/CarRepository.cs [hunk 7]
```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SellCar.DAL.Interfaces;
+using SellCar.Domain.Models;
+using SellsCar.DAL;
+
+namespace SellCar.DAL.Repositories
+{
+    public class CarRepository :IBaseRepository<Car>
+    {
+        private readonly DbContextSellCar _db;
```
Comment (VadzimPapko) [inline SellCar/SellCar.DAL/Repositories/CarRepository.cs:10]: please implement Dispose() for the DB context

File: SellCar/SellCar.DAL/Repositories/CarRepository.cs [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+        public CarRepository(DbContextSellCar db)
+        {
+            _db = db;
+        }
+
+        public async Task Create(Car entity)
```
Comment (VadzimPapko) [inline SellCar/SellCar.DAL/Repositories/CarRepository.cs:17]: for async methods, we add an Async word at the end, for instance `CreateAsync`

**Review by VadzimPapko — APPROVED** [review]
Reviewed. Added notes. Please review.

**Pull request merged**